### DISCUSSION TITLE
Are BlackHole Output devices also readable/usable from with `/dev` too ?

### DISCUSSION THREAD

**Discussion by davidandreoletti** [ExistentialAudio/BlackHole#771] [Q&A] [answered]
Hi,

Here is my use case / setup.

Local music client:
- Some music player
- No support for streaming the music over TCP/UDP/etc

Remote Audio Server:
- Accept any kind of audio stream over TCP/UDP

I would like to bridge the two endpoints as follow:

```
Local music client -> Aggregated Blackhole device ->  cat /dev/aggregated_black_hole  | nc remote-audio-server 3333 -> Speaker
```

Is there a block/char device readable within the /dev file hierarchy for each BlackHole Output devices ?

**Comment by gchilds** [comment, marked as answer]
The libASPL sample device writes to a UDP socket:

https://github.com/gavv/libASPL/blob/main/examples/NetcatDevice/Driver.cpp

**Reply by davidandreoletti** [reply]
Thank you @gchilds 

LibASPL is embedded in ROC VAD (https://github.com/roc-streaming/roc-vad) / its parent project. I will use that to create the socket.

**Reply by gchilds** [reply]
Oh wow, interesting project!
